pkg/rpctype/rpctype.go

```diff
@@ -17,12 +17,23 @@ import (
 // ExecutionRequest describes the task of executing a particular program.
 // Corresponds to Fuzzer.Request.
 type ExecutionRequest struct {
-	ID               int64
-	ProgData         []byte
-	ExecOpts         ipc.ExecOpts
-	NewSignal        bool
+	ID        int64
+	ProgData  []byte
+	ExecOpts  ipc.ExecOpts
+	NewSignal bool
+	// If set, ProgData contains compiled executable binary
+	// that needs to be written to disk and executed.
+	IsBinary bool
+	// If set, fully reset executor state befor executing the test.
+	ResetState bool
+	// If set, collect program output and return in ExecutionResult.Output.
+	ReturnOutput bool
+	// If set, don't fail on program failures, instead return the error in ExecutionResult.Error.
+	ReturnError      bool
 	SignalFilter     signal.Signal
 	SignalFilterCall int
+	// Repeat the program that many times (0 means 1).
+	Repeat int
```

Comment:
How many repeats will we typically do during runtest? If the figure is not too big, I wonder if it's easier to just send the program several times?

A few extra RPC exchanges should be very cheap (especially relative to the time it takes to execute a program on an instrumented kernel) and the code will be much simpler.

Reply:
I also don't like this piece of code. But the problem is that runtest executed these repeated programs in a very specific way, it's not just running it several times. They run on the same proc/env with alternating restart/norestart. It tested that it does not accumulate state.
Maybe we can figure out some semi-equivalent way to run repeated programs w/o fuzzer support, or maybe these repeats are just not too important. But for now I am just trying to slowly progress towards removing Go code from the target.